lib/Twig/Extension/Core.php

```diff
@@ -280,19 +280,36 @@ function twig_cycle($values, $position)
  *
  * @param Twig_Environment                   $env
  * @param Traversable|array|int|float|string $values The values to pick a random item from
+ * @param int|float|null                     $max    Max. value used when $values is an int or float
  *
  * @throws Twig_Error_Runtime when $values is an empty array (does not apply to an empty string which is returned as is)
  *
  * @return mixed A random value from the given sequence
  */
-function twig_random(Twig_Environment $env, $values = null)
+function twig_random(Twig_Environment $env, $values = null, $max = null)
 {
     if (null === $values) {
         return mt_rand();
     }
 
     if (is_int($values) || is_float($values)) {
-        return $values < 0 ? mt_rand($values, 0) : mt_rand(0, $values);
+        if (null === $max) {
+            if ($values < 0) {
+                $max = 0;
+                $min = (int) $values;
```

Comment:
I had a hard time figuring out why `$values` is casted to an `int` here. Can we make it more obvious? I have not tested, but I supposed it works well on all possible inputs.